src/azure-cli-core/azure/cli/core/tests/test_util.py

```diff
@@ -412,6 +413,36 @@ def test_resource_to_scopes(self):
         self.assertEqual(resource_to_scopes('https://managedhsm.azure.com'),
                          ['https://managedhsm.azure.com/.default'])
 
+    @mock.patch("psutil.Process")
+    def test_get_parent_proc_name(self, mock_process_type):
+        process = mock_process_type.return_value
+        parent1 = process.parent.return_value
+        parent2 = parent1.parent.return_value
+        parent3 = parent2.parent.return_value
+
+        # Windows, in a virtual env, launched by pwsh.exe
+        process.name.return_value = "python.exe"
+        parent1.name.return_value = "python.exe"
+        parent2.name.return_value = "cmd.exe"
+        parent3.name.return_value = "pwsh.exe"
+        self.assertEqual(_get_parent_proc_name(), "pwsh.exe")
+
+        # Windows, in a virtual env, launched by powershell.exe
+        parent3.name.return_value = "powershell.exe"
+        self.assertEqual(_get_parent_proc_name(), "powershell.exe")
+
+        # Windows, launched by cmd.exe
+        parent1.name.return_value = "cmd.exe"
+        parent2.name.return_value = "explorer.exe"
+        self.assertEqual(_get_parent_proc_name(), "cmd.exe")
+
+        # Linux
+        process.name.return_value = "python"
+        parent1.name.return_value = "bash"
+        parent2.name.return_value = "init"
+        parent3.name.return_value = "init"
+        self.assertEqual(_get_parent_proc_name(), "bash")
```

Comment:
Out of curiosity, I just want to confirm that Mac and Linux are exactly the same, right?

Reply:
Yes. Mac is also using `bash`, but it doesn't really mater as long as no ancestor process is called `powershell.exe`, `pwsh.exe`. 

However, I think `pwsh` running on Linux can be a more complex scenario. I need to do more testing on that.